app/connector/support/maven-plugin/src/test/java/io/syndesis/connector/support/maven/GenerateConnectorInspectionsMojoTest.java

```diff
@@ -125,6 +130,49 @@ public void generateJavaInspectionWithVariantTest() throws Exception {
         assertThat(enriched.findVariantByMeta("variant", "collection")).get().extracting("specification").isNotEmpty();
     }
 
+    @Test
+    public void validFileAgainstSchema() throws Exception {
+        GenerateConnectorInspectionsMojo mojo = new GenerateConnectorInspectionsMojo();
+
+        InputStream fileStream = getClass().getResourceAsStream("/my-test-connector.json");
+        assertNotNull(fileStream);
+        JsonNode jsonNode = mojo.validateWithSchema("testFile", fileStream);
+        assertNotNull(jsonNode);
+    }
+
+    @Test
+    public void unformedFileAgainstSchema() throws Exception {
+        GenerateConnectorInspectionsMojo mojo = new GenerateConnectorInspectionsMojo();
+
+        InputStream fileStream = getClass().getResourceAsStream("/my-unformed-connector.json");
+        assertNotNull(fileStream);
+        try {
+            JsonNode jsonNode = mojo.validateWithSchema("testFile", fileStream);
+            assertNotNull(jsonNode);
+            fail("Should throw an exception at line 9");
```

Comment:
I find these kinds of tests are better by using AssertJ and exception assertions, see http://joel-costigliola.github.io/assertj/assertj-core-features-highlight.html#exception-assertion